### ISSUE TITLE
Improve Full Text Search script

### ISSUE THREAD

**Issue by enoch85** [nextcloud/vm#494]
It's implemented in the scripts now and replaced Nextant. Though there are some issues left for a perfect install.

fulltextsearch:index doesn't work, and the checks to see if elastic search is running fails if they are not commented out.

Previous issue was here: https://github.com/nextcloud/vm/issues/431

**Comment by enoch85** [comment]
cc @daita @morph027 @Ark74 Please test the current state and report back what isn't working. Maybe even throw in a PR while you're at it. ;)

**Comment by Ark74** [comment]
Interesting. I'll try to take a look. Nextant was definitely simpler to set up :smile:

**Comment by Brailyn** [comment]
Running the latest script gives me:

```
Starting to setup Elastic Search & Full Text Search on Nextcloud...
[\]

  [Symfony\Component\Console\Exception\CommandNotFoundException]
  Command "app:disbale" is not defined.
  Did you mean one of these?
      app:check-code
      app:disable
      app:enable
      app:getpath
      app:install
      app:list
      encryption:disable
      twofactorauth:disable
      user:disable


Sorry but something went wrong. Please report this issue to https://github.com/nextcloud/vm/issues and include the output of the error message. Thank you!
```

**Comment by morph027** [comment]
I see a typo crying out loud, should be `disable` instead of `disbale` ;)

**Comment by enoch85** [comment]
Fixed with https://github.com/nextcloud/vm/commit/b5fb54d26260d6ba64a3f3d3cf8aaf61ce6cc564

**Comment by Brailyn** [comment]
Install works better now!

> fulltextsearch:index doesn't work, and the checks to see if elastic search is running fails if they are not commented out.

This is where the script gets stuck... Do you know how I can fix this in the mean time?

```
  [OCA\FullTextSearch_ElasticSearch\Exceptions\ConfigurationException]
  Your ElasticSearchPlatform is not configured properly


fulltextsearch:index [--output [OUTPUT]]

Sorry but something went wrong. Please report this issue to https://github.com/nextcloud/vm/issues and include the output of the error message. Thank you!
sudo -u www-data php /var/www/nextcloud/occ fulltextsearch:index failed
```

Not sure if it helps, but I noticed in the web gui, the **Address of the Servlet** and **Index** are blank.

**Comment by enoch85** [comment]
@Brailyn I'd be happy to assist you if you can find a solution and send a PR to this repo. Seems like both @Ark74 and @morph027 didn't have the time to look at this yet, either did I.

**Comment by daita** [comment]
Yes, the exception `Your ElasticSearchPlatform is not configured properly` means that one of the field is empty.

Could you please paste the result of `./occ fulltextsearch:check` ?

**Comment by Brailyn** [comment]
```
ncadmin@nextcloud:~$ sudo -u www-data php /var/www/nextcloud/occ fulltextsearch:check
Full text search 0.6.1

- Search Platform:
Elasticsearch 0.6.1
{
    "elastic_host": "http:\/\/:",
    "elastic_index": "nextcloud"
}

- Content Providers:
Files 0.6.0
{
    "files_local": "1",
    "files_external": "0",
    "files_encrypted": "0",
    "files_federated": "0",
    "files_size": "20",
    "files_pdf": "1",
    "files_office": "1",
    "files_image": "0",
    "files_audio": "0"
}
```

```
ncadmin@nextcloud:~$ sudo bash /etc/init.d/elasticsearch start
[ ok ] Starting elasticsearch (via systemctl): elasticsearch.service.
ncadmin@nextcloud:~$ curl http://localhost:9200
curl: (7) Failed to connect to localhost port 9200: Connection refused
```

**Comment by enoch85** [comment]
@daita Could you help out here please?

**Comment by Ark74** [comment]
@Brailyn loose the http or better 
`netstat -atunlp | grep 9200`

**Comment by Ark74** [comment]
@daita Hi!
Quick question do we need to stick to a specific release within 6.1.x or a specific one like _6.1.1_ as recommended on the fulltextsearch wiki?
The VM Script uses version 6.1.1 (right, @enoch85) and i'm almost sure that some plugins require that version to install, so maybe we might need to use `apt-mark hold` in order to prevent some breakage while updating.

What do you think?

**Comment by daita** [comment]
No, you can use the last version of elasticsearch, it should be fine. However, the plugin from readonlyrest might takes some day to be update after a fresh release of elasticsearch

**Comment by enoch85** [comment]
@Ark74 

> The VM Script uses version 6.1.1 

Would be nice to automate the download of the plugin. i didn't find a public repository yet though. Maybe we could use some of the old functions to get the latest version in that case? My current "repo" is this: https://github.com/nextcloud/vm/tree/master/apps/fulltextsearch-files Putting files there and manually changing the version will work, but we don't like manual processes, right? ;)

**Comment by Ark74** [comment]
>  i didn't find a public repository yet though
What about
https://github.com/sscarduzio/elasticsearch-readonlyrest-plugin/releases

Maybe filtering the results?

**Comment by nxtester** [comment]
Hello,
today I installed a fresh NC machine, based on the latest script. All went fine, installation was a breeze.

Fulltext isn't working:

![grafik](https://user-images.githubusercontent.com/28905767/40137383-92c85c56-594a-11e8-8c61-c5d4cc8fab39.png)

Any idea?

**Comment by Ark74** [comment]
There are still issues, hence this issue is still open.
Any contribution or PR is welcome.

El 16 de mayo de 2018 1:50:10 PM CDT, nxtester <notifications@github.com> escribió:
>Hello,
>today I installed a fresh NC machine, based on the latest script. All
>went fine, installation was a breeze.
>
>Fulltext isn't working:
>
>![grafik](https://user-images.githubusercontent.com/28905767/40137383-92c85c56-594a-11e8-8c61-c5d4cc8fab39.png)
>
>Any idea?
>
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/nextcloud/vm/issues/494#issuecomment-389626501

-- 
Enviado desde dispositivo móvil. Por favor, disculpa mi brevedad.

**Comment by enoch85** [comment]
> What about
https://github.com/sscarduzio/elasticsearch-readonlyrest-plugin/releases
Maybe filtering the results?

Hmm, missed that. Too obvious hehe.

Yeah, always get the latest plugin and install it with compatible a elasticsearch version.

**Comment by Ark74** [comment]
> Yeah, always get the latest plugin and install it with compatible a elasticsearch version.

Yeah I've seen that as a standard that's why I was wondering about it :laughing:

**Comment by Ark74** [comment]
Hopefully https://github.com/nextcloud/vm/pull/541/commits/52e4867bfd019d9aa7156f33e8b67201d7976ce2 will close this one :smiley: 
Cheers!

**Comment by enoch85** [comment]
Merged. Can anyone test?
